README.md

```diff
@@ -1,2 +1,226 @@
+├── public
+│   ├── images
+│   ├── javascripts
+│   └── stylesheets
+│       └── style.css
+├── routes
+│   ├── index.js
+│   └── users.js
+└── views
+    ├── error.ejs
+    └── index.ejs
+```
+
+Since we are working on backend server for ecommerce application this structure is updated as follow:
+
+```
+.
+├── app.js
+├── bin
+│   └── www
+├── config
+├── models
+├── node_modules
+├── package.json
+├── package-lock.json
+├── routes
+│   ├── api
+│   │   ├── index.js
+│   │   └── v1
+│   │       └── index.js
+│   └── index.js
+├── src
+│   ├── constants
+│   ├── controller
+│   ├── middlewares
+│   ├── models
+│   └── services
+├── tests
+└── views
+    ├── error.ejs
+    └── index.ejs
+
+
+```
+
+
+## Database Setup
+
+### 1. MySQL Installation:
+
+>LINUX
+
+```sh
+sudo apt update
+sudo apt install mysql-server
+```
+
+> Windows
+
+Follow [this](https://dev.mysql.com/doc/refman/8.0/en/windows-installation.html) to download the mysql on windows.
+
+> MAC
+
+```sh
+brew install mysql
+```
+
+
+### 2. MySQL Configuration:
+
+In Linux the root user of mysql is set to perform the authentication using the `auth_socket` by default. You can check it by running the following command:
+```sh
+ sudo mysql
+ mysql> SELECT user,authentication_string,plugin,host FROM mysql.user;
+```
+
+
+| user | authentication_string|auth_socket | localhost|
+| :---: | :---: | :---: | :---: | 
+| root             |                                           | auth_socket           | localhost |
+| mysql.session    | *THISISNOTAVALIDPASSWORDTHATCANBEUSEDHERE | mysql_native_password | localhost |
+| mysql.sys        | *THISISNOTAVALIDPASSWORDTHATCANBEUSEDHERE | mysql_native_password | localhost |
+| debian-sys-maint | *70FC20C6FDEF784AA4BEACA2F4BFF67F5B228C32 | mysql_native_password | localhost |
+4 rows in set (0.00 sec)
+
+To allow our ecommerce application connect to this database we need to set the password for the root. 
+```sh
+mysql> ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY 'password';
+```
+Now we need to make run the following command to make the changes takes place
+```sh
+mysql> FLUSH PRIVILEGES;
+```
+ and thats it. To test if the authentication using the password is working or not we will getout of the mysql and retry logging in using the user as `root` and password as `password`
+ 
+ ```sh
+mysql> exit;
+> mysql -u root -p
+> password
+ ```
+
+Create a Database `ecommercedb` for this application using the following command:
+```sh
+CREATE DATABASE ecommercedb;
+```
+
+Since we will be using the database, run the following command
+```sh
+USE ecommercedb;
+```
+
+## Table Creation:
+
+Use the following command to create the tables :
+
+> Categories
+
+```
+CREATE TABLE `Categories` (
+  `ID` int NOT NULL AUTO_INCREMENT,
+  `Name` varchar(255) NOT NULL,
+  `CreatedAt` datetime DEFAULT CURRENT_TIMESTAMP,
+  `UpdatedAt` datetime DEFAULT CURRENT_TIMESTAMP,
+  PRIMARY KEY (`ID`)
+) ENGINE=InnoDB AUTO_INCREMENT=1 DEFAULT CHARSET=utf8mb4;;
```

Comment:
There are two semicolons at the end of this query, which creates an empty query. Please fix this.